src/main/java/spleetwaise/address/model/AddressBook.java

```diff
@@ -76,6 +77,17 @@ public Optional<Person> getPersonByPhone(Phone phone) {
         return Optional.of(filteredPersonList.get(0));
     }
 
+    public Optional<Person> getPersonByIndex(Index index) {
+        requireNonNull(index);
+
+        ObservableList<Person> PersonList = getPersonList();
```

Comment:
The index the user uses is should be based on the filtered list, thus this might not be fetching the correct person.